x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/action/TransportGetLifecycleAction.java

```diff
@@ -49,32 +50,37 @@ protected Response newResponse() {
     }
 
     @Override
-    protected void masterOperation(Request request, ClusterState state, ActionListener<Response> listener) throws Exception {
+    protected void masterOperation(Request request, ClusterState state, ActionListener<Response> listener) {
         IndexLifecycleMetadata metadata = clusterService.state().metaData().custom(IndexLifecycleMetadata.TYPE);
         if (metadata == null) {
             listener.onFailure(new ResourceNotFoundException("Lifecycle policy not found: {}", Arrays.toString(request.getPolicyNames())));
         } else {
-            List<LifecyclePolicy> requestedPolicies;
+            List<LifecyclePolicyMetadata> requestedPolicies;
             // if no policies explicitly provided, behave as if `*` was specified
             if (request.getPolicyNames().length == 0) {
-                requestedPolicies = new ArrayList<>(metadata.getPolicies().values());
+                requestedPolicies = new ArrayList<>(metadata.getPolicyMetadatas().values());
             } else {
                 requestedPolicies = new ArrayList<>(request.getPolicyNames().length);
                 for (String name : request.getPolicyNames()) {
-                    LifecyclePolicy policy = metadata.getPolicies().get(name);
-                    if (policy == null) {
+                    LifecyclePolicyMetadata policyMetadata = metadata.getPolicyMetadatas().get(name);
+                    if (policyMetadata == null) {
                         listener.onFailure(new ResourceNotFoundException("Lifecycle policy not found: {}", name));
                         return;
                     }
-                    requestedPolicies.add(policy);
+                    requestedPolicies.add(policyMetadata);
                 }
             }
-            listener.onResponse(new Response(requestedPolicies));
+            List<LifecyclePolicyResponseItem> responseItems = new ArrayList<>(requestedPolicies.size());
+            for (LifecyclePolicyMetadata policyMetadata : requestedPolicies) {
+                responseItems.add(new LifecyclePolicyResponseItem(policyMetadata.getPolicy(),
+                    policyMetadata.getVersion(), policyMetadata.getCreationDateString()));
+            }
```

Comment:
Can we not create the list of `LifecyclePolicyResponseItem` directly rather than first creating the list of `LifecyclePolicyMetadata`?